### ISSUE TITLE
Lossy Artifacts

### ISSUE THREAD

**Issue by MrFanservice-png** [VoxelCubes/PanelCleaner#102]
Whenever I use the Inpainter or set the mask max standard deviation to the max, the output files have artifacts in them. How do I lossless remove text from manga with having any artifacts?

2024-07-12 16:42:21.128 | INFO     | pcleaner.gui.launcher:launch:100 - 
- Program Information -
Program: Panel Cleaner 2.7.0
Executing from: C:\Users\<username>\Documents\PanelCleaner\_internal\pcleaner\gui\launcher.pyc
Log file: C:\Users\<username>\AppData\Roaming\pcleaner\cache\pcleaner.log
Config file: C:\Users\<username>\AppData\Roaming\pcleaner\pcleanerconfig.ini
Cache directory: C:\Users\<username>\AppData\Roaming\pcleaner\cache
- System Information -
Operating System: Windows 10
Machine: AMD64
Python Version: 3.11.0 (main, Oct 24 2022, 18:26:48) [MSC v.1933 64 bit (AMD64)]
PySide (Qt) Version: 6.7.0
Available Qt Themes: windows11, windowsvista, Windows, Fusion
System locale: en_US
CPU Cores: 12
GPU: None (CUDA not available)

2024-07-12 16:42:21.199 | INFO     | pcleaner.gui.launcher:launch:114 - Using locale en_US.
2024-07-12 16:42:21.201 | DEBUG    | pcleaner.gui.launcher:launch:121 - Loaded built-in Qt translations for en_US.
2024-07-12 16:42:21.202 | DEBUG    | pcleaner.gui.launcher:launch:129 - Loaded built-in Qt base translations for en_US.
2024-07-12 16:42:21.378 | DEBUG    | pcleaner.gui.mainwindow_driver:ensure_models_downloaded:351 - Text detector model already downloaded.
2024-07-12 16:42:21.379 | DEBUG    | pcleaner.gui.mainwindow_driver:initialize_ui:210 - Purging missing profiles.
2024-07-12 16:42:21.379 | INFO     | pcleaner.gui.mainwindow_driver:initialize_profiles:863 - Found profiles: [('Default', None), ('pp', WindowsPath('C:/Users/<username>/AppData/Roaming/pcleaner/profiles/pp.conf')), ('qw', WindowsPath('C:/Users/<username>/AppData/Roaming/pcleaner/profiles/qw.conf'))]
2024-07-12 16:42:21.380 | DEBUG    | pcleaner.config:load_profile:1152 - Loading profile None...
2024-07-12 16:42:21.380 | DEBUG    | pcleaner.config:load_profile:1159 - Loading builtin default profile
2024-07-12 16:42:21.467 | DEBUG    | pcleaner.gui.mainwindow_driver:load_current_profile:1021 - Loading current profile.
2024-07-12 16:42:21.467 | DEBUG    | pcleaner.gui.profile_parser:set_profile_values:432 - Setting profile values
2024-07-12 16:42:21.471 | DEBUG    | pcleaner.gui.mainwindow_driver:initialize_analytics_view:630 - Loading included font from C:\Users\<username>\Documents\PanelCleaner\_internal\pcleaner\data\NotoMono-Regular.ttf
2024-07-12 16:42:21.472 | DEBUG    | pcleaner.gui.mainwindow_driver:initialize_analytics_view:633 - Loaded included font
2024-07-12 16:42:21.474 | DEBUG    | pcleaner.gui.mainwindow_driver:save_default_palette:129 - Placeholder color: #000000
2024-07-12 16:42:21.479 | INFO     | pcleaner.gui.mainwindow_driver:set_theme:148 - Using system theme.
2024-07-12 16:42:21.479 | INFO     | pcleaner.gui.mainwindow_driver:changeEvent:198 - Theme is dark: False
2024-07-12 16:42:21.481 | INFO     | pcleaner.gui.mainwindow_driver:set_theme:169 - Theme is dark: False
2024-07-12 16:42:21.628 | INFO     | pcleaner.gui.mainwindow_driver:changeEvent:198 - Theme is dark: False
2024-07-12 16:42:21.632 | DEBUG    | pcleaner.gui.mainwindow_driver:post_init:412 - Char width: 6, columns: 74, required width: 444
2024-07-12 16:42:21.634 | DEBUG    | pcleaner.gui.mainwindow_driver:post_init:443 - Splitter sizes: [375, 423, 463]
2024-07-12 16:42:21.635 | DEBUG    | pcleaner.gui.mainwindow_driver:start_initialization_worker:561 - Worker Thread cleaning cache
2024-07-12 16:42:21.635 | DEBUG    | pcleaner.gui.mainwindow_driver:start_initialization_worker:568 - Worker Thread loading OCR model.
2024-07-12 16:42:21.636 | INFO     | pcleaner.ocr.ocr_mangaocr:__new__:15 - Creating the MangaOcr instance
2024-07-12 16:42:21.636 | ERROR    | pcleaner.ocr.ocr_tesseract:available_langs:24 - Error checking Tesseract available language data: tesseract is not installed or it's not in your PATH. See README file for more information.
2024-07-12 16:42:21.637 | ERROR    | pcleaner.ocr.ocr_tesseract:available_langs:24 - Error checking Tesseract available language data: tesseract is not installed or it's not in your PATH. See README file for more information.
2024-07-12 16:42:21.637 | INFO     | manga_ocr.ocr:__init__:13 - Loading OCR model from kha-white/manga-ocr-base
2024-07-12 16:42:23.469 | INFO     | manga_ocr.ocr:__init__:25 - Using CPU
2024-07-12 16:42:24.037 | INFO     | manga_ocr.ocr:__init__:32 - OCR ready
2024-07-12 16:42:24.037 | INFO     | pcleaner.gui.mainwindow_driver:load_ocr_model:585 - Loaded OCR model (2.40s)
2024-07-12 16:42:30.961 | DEBUG    | pcleaner.gui.file_table:handleDrop:197 - Dropped C:/Users/<username>/Downloads/118.png
2024-07-12 16:42:30.962 | DEBUG    | pcleaner.gui.file_table:add_file:246 - Requesting to add "C:\Users\<username>\Downloads\118.png"
2024-07-12 16:42:30.962 | DEBUG    | pcleaner.gui.file_table:repopulate_table:303 - Repopulating table
2024-07-12 16:42:30.965 | INFO     | pcleaner.gui.file_table:lazy_load_images:619 - Dispatching image loading workers
2024-07-12 16:42:30.965 | DEBUG    | pcleaner.gui.file_table:lazy_load_images:625 - Worker Thread loading image C:\Users\<username>\Downloads\118.png
2024-07-12 16:42:48.319 | WARNING  | pcleaner.gui.mainwindow_driver:profile_change_check:1033 - Profile change check
2024-07-12 16:42:48.319 | DEBUG    | pcleaner.config:load_profile:1152 - Loading profile 'pp'...
2024-07-12 16:42:48.320 | DEBUG    | pcleaner.config:load_profile:1166 - Loading profile pp from C:\Users\<username>\AppData\Roaming\pcleaner\profiles\pp.conf
2024-07-12 16:42:48.320 | INFO     | pcleaner.config:load:862 - Loading profile C:\Users\<username>\AppData\Roaming\pcleaner\profiles\pp.conf from disk...
2024-07-12 16:42:48.321 | DEBUG    | pcleaner.gui.mainwindow_driver:load_current_profile:1021 - Loading current profile.
2024-07-12 16:42:48.321 | DEBUG    | pcleaner.gui.profile_parser:set_profile_values:432 - Setting profile values
2024-07-12 16:43:20.809 | INFO     | pcleaner.gui.mainwindow_driver:start_cleaning:1317 - Requested outputs: [<Output.denoised_output: 19>, <Output.write_output: 22>]
2024-07-12 16:43:20.811 | INFO     | pcleaner.gui.processing:generate_output:176 - Running text detection AI model for 1 images...
2024-07-12 16:43:22.303 | DEBUG    | pcleaner.ctd_interface:process_image:195 - Saving json file to C:\Users\<username>\AppData\Roaming\pcleaner\cache\cleaner\4288d7e4-db98-4aa6-b0c6-a337f9c944ba_118#raw.json
2024-07-12 16:43:22.308 | DEBUG    | pcleaner.image_ops:visualize_raw_boxes:894 - Loading included font from C:\Users\<username>\Documents\PanelCleaner\_internal\pcleaner\data\LiberationSans-Regular.ttf
2024-07-12 16:43:22.510 | INFO     | pcleaner.gui.processing:generate_output:228 - Running preprocessing for 1 images...
2024-07-12 16:43:22.510 | DEBUG    | pcleaner.preprocessor:prep_json_file:120 - Processing json file: C:\Users\<username>\AppData\Roaming\pcleaner\cache\cleaner\4288d7e4-db98-4aa6-b0c6-a337f9c944ba_118#raw.json
2024-07-12 16:43:22.511 | DEBUG    | pcleaner.preprocessor:prep_json_file:161 - Detected lang: eng
2024-07-12 16:43:22.515 | INFO     | pcleaner.gui.processing:generate_output:290 - Running masker for 1 images...
2024-07-12 16:43:22.515 | INFO     | pcleaner.gui.mainwindow_driver:show_current_progress:1529 - Showing ocr analytics...
2024-07-12 16:43:23.104 | INFO     | pcleaner.gui.processing:generate_output:404 - Running denoiser for 1 images...
2024-07-12 16:43:23.104 | INFO     | pcleaner.gui.mainwindow_driver:show_current_progress:1538 - Showing masker analytics...
2024-07-12 16:43:23.343 | INFO     | pcleaner.gui.processing:generate_output:602 - Finished processing 1 images.
2024-07-12 16:43:23.343 | INFO     | pcleaner.gui.mainwindow_driver:show_current_progress:1548 - Showing denoiser analytics...
2024-07-12 16:43:23.377 | DEBUG    | pcleaner.image_ops:save_optimized:858 - Saving image 118_clean.png with kwargs: {'optimize': True, 'compress_level': 9, 'dpi': (72.009, 72.009)}
2024-07-12 16:43:25.569 | INFO     | pcleaner.gui.mainwindow_driver:output_worker_result:1461 - Output worker finished.
2024-07-12 16:43:28.032 | WARNING  | pcleaner.gui.mainwindow_driver:profile_change_check:1033 - Profile change check
2024-07-12 16:43:28.033 | DEBUG    | pcleaner.config:load_profile:1152 - Loading profile 'qw'...
2024-07-12 16:43:28.033 | DEBUG    | pcleaner.config:load_profile:1166 - Loading profile qw from C:\Users\<username>\AppData\Roaming\pcleaner\profiles\qw.conf
2024-07-12 16:43:28.033 | INFO     | pcleaner.config:load:862 - Loading profile C:\Users\<username>\AppData\Roaming\pcleaner\profiles\qw.conf from disk...
2024-07-12 16:43:28.035 | DEBUG    | pcleaner.gui.mainwindow_driver:load_current_profile:1021 - Loading current profile.
2024-07-12 16:43:28.035 | DEBUG    | pcleaner.gui.profile_parser:set_profile_values:432 - Setting profile values
2024-07-12 16:43:34.860 | INFO     | pcleaner.gui.mainwindow_driver:start_cleaning:1317 - Requested outputs: [<Output.inpainted_output: 21>, <Output.write_output: 22>]
2024-07-12 16:43:34.861 | INFO     | pcleaner.gui.processing:generate_output:290 - Running masker for 1 images...
2024-07-12 16:43:35.120 | INFO     | pcleaner.gui.processing:generate_output:404 - Running denoiser for 1 images...
2024-07-12 16:43:35.120 | INFO     | pcleaner.gui.mainwindow_driver:show_current_progress:1538 - Showing masker analytics...
2024-07-12 16:43:35.284 | INFO     | pcleaner.gui.processing:generate_output:503 - Running inpainting for 1 images...
2024-07-12 16:43:35.284 | INFO     | pcleaner.gui.mainwindow_driver:show_current_progress:1548 - Showing denoiser analytics...
2024-07-12 16:43:42.943 | INFO     | pcleaner.gui.processing:generate_output:602 - Finished processing 1 images.
2024-07-12 16:43:42.943 | INFO     | pcleaner.gui.mainwindow_driver:show_current_progress:1561 - Showing inpainter analytics...
2024-07-12 16:43:42.982 | DEBUG    | pcleaner.image_ops:save_optimized:858 - Saving image 118_clean.png with kwargs: {'optimize': True, 'compress_level': 9, 'dpi': (72.009, 72.009)}
2024-07-12 16:43:45.933 | INFO     | pcleaner.gui.mainwindow_driver:output_worker_result:1461 - Output worker finished.
2024-07-12 16:44:00.777 | DEBUG    | pcleaner.gui.mainwindow_driver:open_issue_reporter:760 - Opening issue reporter.

![118](https://github.com/user-attachments/assets/c1d922da-96ba-4f98-868f-8f5c1f73cd21)
![118_clean](https://github.com/user-attachments/assets/3207430e-927f-47ca-84a3-c20743567258)

**Comment by VoxelCubes** [comment]
You can't. Inpainting can cause artifacts, hence being disabled by default. Setting the max deviation super high also causes artifacts, hence the default being lower. The result is an artifact-free cleaning with the exception of a few bubbles that Panel Cleaner simply cannot clean well by itself, without causing a few artifacts. 

You just need to deal with those manually. 
Machine Learning AI is not perfect. It will not deliver perfection. Please stop expecting it to be perfect, it simply cannot be done.
You must clean up the few remaining bits of text yourself. How many bits are left can be carefully tweaked with the settings. This varies depending on the manga. I will not help you find the optimal settings, use the defaults and adjust as you see fit from there.

To sum it up: the tool isn't perfect. Go back to default settings and clean the tricky left over bits by hand.

I'm sorry, there is nothing more I can do for you.
Panel Cleaner can do a lot of work for you, but not all of it. I hope you can accept that.
